src/ifcwrap-csharp/IfcOpenShell.Net/IfcOpenShell.Net/PsetExtensions.cs

```diff
@@ -0,0 +1,374 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace IfcOpenShell
+{
+    public static class PsetExtensions
+    {
+        public static IReadOnlyDictionary<string, ArgumentResult>? GetPset(this EntityInstance element,
+            string psetName, 
+            string? propertyName = null,
+            bool onlyPsets = false, 
+            bool onlyQtos = false,
+            bool shouldInherit = true,
+            bool verbose = false
+        )
+        {
+            EntityInstance? pset = null;
+            IReadOnlyDictionary<string, ArgumentResult>? typePset = null;
+            if (element.Is("IfcTypeObject") && 
```

Comment:
Add a comment that his is copyied from python and rerference the file it was copied from